java-tests/validation-gen/src/test/java/io/spine/test/tools/validate/setonce/SetOnceFieldsTest.java

```diff
@@ -0,0 +1,811 @@
+                    .setHeight(halfOfMeter)
+                    .setHeight(halfOfMeter)
+                    .build());
+        }
+
+        @Test
+        @DisplayName("by reflection")
+        void byReflection() {
+            assertValidationPasses(() -> unheightedStudent.toBuilder()
+                    .setField(Student.getDescriptor().findFieldByName("height"), halfOfMeter)
+                    .setField(Student.getDescriptor().findFieldByName("height"), halfOfMeter)
+                    .build());
+        }
+
+        @Test
+        @DisplayName("by message merge")
+        void byMessageMerge() {
+            assertValidationPasses(() -> unheightedStudent.toBuilder()
+                    .mergeFrom(shortStudent)
+                    .mergeFrom(shortStudent)
+                    .build());
+        }
+
+        @Test
+        @DisplayName("by bytes merge")
+        void byBytesMerge() {
+            assertValidationPasses(() -> unheightedStudent.toBuilder()
+                    .mergeFrom(shortStudent.toByteArray())
+                    .mergeFrom(shortStudent.toByteArray())
+                    .build());
+        }
+
+        @Test
+        @DisplayName("after clearing")
+        void afterClearing() {
+            assertValidationPasses(() -> shortStudent.toBuilder()
+                    .clearHeight()
+                    .setHeight(halfOfMeter)
+                    .build());
+        }
+    }
+
+    @Nested
+    @DisplayName("prohibit overriding non-default float")
+    class ProhibitOverridingNonDefaultFloat {
+
+        private static final float fiftyKilograms = 0.5f;
+
+        private final Student heavyStudent = Student.newBuilder()
+                .setWeight(80)
+                .build();
+        private final Student thinStudent = Student.newBuilder()
+                .setWeight(fiftyKilograms)
+                .build();
+
+        @Test
+        @DisplayName("by value")
+        void byValue() {
+            assertValidationFails(() -> heavyStudent.toBuilder()
+                    .setWeight(fiftyKilograms));
+        }
+
+        @Test
+        @DisplayName("by reflection")
+        void byReflection() {
+            assertValidationFails(() -> heavyStudent.toBuilder()
+                    .setField(Student.getDescriptor().findFieldByName("weight"), fiftyKilograms));
+        }
+
+        @Test
+        @DisplayName("by message merge")
+        void byMessageMerge() {
+            assertValidationFails(() -> heavyStudent.toBuilder()
+                    .mergeFrom(thinStudent));
+        }
+
+        @Test
+        @DisplayName("by bytes merge")
+        void byBytesMerge() {
+            assertValidationFails(() -> heavyStudent.toBuilder()
+                    .mergeFrom(thinStudent.toByteArray()));
+        }
+    }
+
+    @Nested
+    @DisplayName("allow overriding default and same-value float")
+    class AllowOverridingDefaultAndSameValueFloat {
+
+        private static final float fiftyKilograms = 0.5f;
```

Comment:
I would have these repeating values moved to a corresponding test environment.

It would simplify making any changes, such as renaming, or changing the value.